src/queue.js

```diff
@@ -89,11 +89,17 @@ export default class Queue extends EventEmitter {
       }
 
       const runAgain = (items) => {
-        if (items.length < self.options.concurrency && options.oneShot) {
-          return Bluebird.resolve()
+        if (options.oneShot) {
+          if (items.length < self.options.concurrency) {
+            return Bluebird.resolve()
+          }
+
+          return pollItems()
```

Comment:
Pq essa alteração? 🤔

Reply:
Actually that's a new line 😬 

Thats due the nature of the `oneShot` expected behavior.
This should act like a full process of the available messages at the queue, and after processing all messages it will resolve the promise and return to the caller.

That's mainly used at tests and because it waits the promise to resolve we don't have to use dalays and sometimes add intermittency to the tests suite.